### DISCUSSION TITLE
Is request `Uri` guaranteed to be without host?

### DISCUSSION THREAD

**Discussion by nakedible-p** [tokio-rs/axum#2644] [Q&A]
### Summary

Is the `Uri` (and `OriginalUri`) guaranteed to always be without a path in the request? If so, where is this documented? I'd hate to rely on something that's more of an accident than intentional design.

Or is it possible it will have a scheme and/or authority?

### axum version

0.7.4

**Comment by jplatte** [comment]
This is a question for [`hyper`](https://github.com/hyperium/hyper), axum does nothing to the request's URI, other than making it available as an extractor.
